src/core/packet.lua

```diff
@@ -20,10 +18,40 @@ local header_size = 8
 local max_payload = tonumber(C.PACKET_PAYLOAD_SIZE)
 
 -- Freelist containing empty packets ready for use.
+
+ffi.cdef[[
+struct freelist {
+    int nfree, max;
+    struct packet *list[?];
+};
+]]
```

Comment:
`for i in {1..10}; do sudo ./bench/basic1-100e6; done`
Without the patch:
```
27.6
29.5
29.4
29.4
29.0
29.2
29.4
29.4
29.3
28.5
```
With the patch:
```
28.3
30.0
29.9
29.8
30.0
29.8
29.3
30.0
30.0
29.9
```

I know, it's not much. But the basic1 benchmark has only 3 links to process. I believe the effect can (and will) be multiplied once you scale your configuration.